src/tr1/game/level.c

```diff
@@ -451,20 +459,31 @@ static void M_LoadSprites(VFILE *file)
 
     m_LevelInfo.sprite_count = VFile_ReadS32(file);
     for (int i = 0; i < m_LevelInfo.sprite_count; i++) {
-        const GAME_OBJECT_ID object_id = VFile_ReadS32(file);
+        const int32_t object_id = VFile_ReadS32(file);
         const int16_t num_meshes = VFile_ReadS16(file);
         const int16_t mesh_idx = VFile_ReadS16(file);
 
-        if (object_id < O_NUMBER_OF) {
-            OBJECT *object = &g_Objects[object_id];
+        if (object_id >= 0 && object_id < O_NUMBER_OF) {
+            OBJECT *const object = &g_Objects[object_id];
             object->mesh_count = num_meshes;
             object->mesh_idx = mesh_idx;
             object->loaded = 1;
         } else if (object_id - O_NUMBER_OF < STATIC_NUMBER_OF) {
-            STATIC_INFO *object = &g_StaticObjects[object_id - O_NUMBER_OF];
-            object->mesh_count = num_meshes;
-            object->mesh_num = mesh_idx;
-            object->loaded = true;
+            STATIC_INFO *const object =
+                &g_StaticObjects[object_id - O_NUMBER_OF];
+            if (object->loaded) {
+                LOG_WARNING(
+                    "sprite %d is already loaded "
+                    "(trying to override %d:%d with %d:%d)",
+                    object_id - O_NUMBER_OF, object_id - O_NUMBER_OF,
```

Comment:
I think there is an extra `object_id - O_NUMBER_OF` here

Reply:
Good catch – fixed